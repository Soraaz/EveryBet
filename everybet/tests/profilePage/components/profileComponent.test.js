import React from "react";
import renderer from "react-test-renderer";
import {shallow} from "enzyme/build";

import ProfileComponent from "../../../src/profilePage/components/profileComponent";
import User from "../../../src/classes/user";
import Button from "../../../src/common/components/button";

class MockNotification{
    checkToken = jest.fn();
}

describe("ProfileComponent", () => {
    User.notificationCenter = new MockNotification();

	describe("rendering", () => {
		it("Should render correctly", () => {
			const tree = renderer
				.create(<ProfileComponent />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly with a custom user avatar", () => {
			User.avatar = "avatar";
			const tree = renderer
				.create(<ProfileComponent />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});
	});

	describe("functionalities", () => {
		const navigation = {navigate: jest.fn(x => x)};
		let wrapper;

		beforeEach(() => {
			wrapper = shallow(<ProfileComponent navigation={navigation} />);
		});

		it("Should navigate to EditProfile when pressing \"Editer le profil\"", () => {
			expect(wrapper.find(Button).first().props().onPress()).to.be.equal('EditProfile');
		});

		it("Should navigate to BetProposal when pressing \"Proposer un pari\"", () => {
			expect(wrapper.find(Button).at(1).props().onPress()).to.be.equal('BetProposal');
		});

		it("Should navigate to UserBets when pressing \"Paris en cours\"", () => {
			expect(wrapper.find(Button).at(2).props().onPress()).to.be.equal('UserBets');
		});

		it("Should navigate to UserBets when pressing \"Historique des paris\"", () => {
			expect(wrapper.find(Button).at(3).props().onPress()).to.be.equal('UserBets');
		});

		it("Should navigate to Contact when pressing \"Contacter Everybet\"", () => {
			expect(wrapper.find(Button).at(5).props().onPress()).to.be.equal('Contact');
		});

		it("Should call the callback _disconnect when pressing \"Se dÃ©connecter\"", () => {
			const spy = sinon.spy(wrapper.instance(), "_disconnect");
			wrapper.find(Button).last().props().onPress();
			expect(spy).to.be.calledOnce;
		});

		it("Should disconnect the user", async () => {
			let userInfos =
				{
					id: 0,
					login: "login",
					firstName: "firstName",
					email: "email",
					coins: 0,
					address: null,
					additionalInformation: null,
					zipCode: null,
					country: null,
					phone: null,
					avatar: "",
					reward: false,
					rewardTier: 1,
					rewardValue: 10
				};
			User.setUser(userInfos);
			await wrapper.instance()._disconnect();
			expect(User.id).to.be.equal(undefined);
			expect(User.login).to.be.equal(undefined);
			expect(User.firstName).to.be.equal(undefined);
			expect(User.coins).to.be.equal(undefined);
			expect(User.email).to.be.equal(undefined);
			expect(User.address).to.be.equal(undefined);
			expect(User.additionalInformation).to.be.equal(undefined);
			expect(User.zipCode).to.be.equal(undefined);
			expect(User.country).to.be.equal(undefined);
			expect(User.phone).to.be.equal(undefined);
			expect(User.avatar).to.be.equal(undefined);
			expect(User.reward).to.be.equal(false);
			expect(User.rewardTier).to.be.equal(undefined);
			expect(User.rewardValue).to.be.equal(undefined);
		});
	});
});