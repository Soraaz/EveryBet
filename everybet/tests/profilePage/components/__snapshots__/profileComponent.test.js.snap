// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileComponent rendering Should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#F4F4FE",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/user_placeholder.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 65,
          "height": 125,
          "width": 125,
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "System",
          "fontSize": 32,
          "fontWeight": "600",
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginBottom": 10,
          "marginLeft": 45,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontFamily": "System",
              "fontSize": 20,
              "fontWeight": "600",
              "marginTop": 25,
              "textAlign": "center",
              "textAlignVertical": "center",
            }
          }
        >
          BetCoins
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "flexDirection": "row",
              "height": 20,
              "justifyContent": "flex-start",
              "width": 90,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "System",
                "fontSize": 10,
                "marginLeft": "auto",
                "marginRight": 5,
                "textAlign": "right",
              }
            }
          />
        </View>
      </View>
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/betcoin_icon.png",
          }
        }
        style={
          Object {
            "height": 40,
            "marginLeft": 5,
            "marginTop": 30,
            "width": 40,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Editer le profil
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Proposer un pari
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Paris en cours
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Historique des paris
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Regarder une pub pour gagner des pièces
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Contacter Everybet
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FC8D8D",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Se déconnecter
      </Text>
    </View>
  </View>
</View>
`;

exports[`ProfileComponent rendering Should render correctly with a custom user avatar 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#F4F4FE",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <Image
      source={
        Object {
          "method": "GET",
          "uri": "avatar",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 65,
          "height": 125,
          "width": 125,
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "System",
          "fontSize": 32,
          "fontWeight": "600",
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginBottom": 10,
          "marginLeft": 45,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontFamily": "System",
              "fontSize": 20,
              "fontWeight": "600",
              "marginTop": 25,
              "textAlign": "center",
              "textAlignVertical": "center",
            }
          }
        >
          BetCoins
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "flexDirection": "row",
              "height": 20,
              "justifyContent": "flex-start",
              "width": 90,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "System",
                "fontSize": 10,
                "marginLeft": "auto",
                "marginRight": 5,
                "textAlign": "right",
              }
            }
          />
        </View>
      </View>
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/betcoin_icon.png",
          }
        }
        style={
          Object {
            "height": 40,
            "marginLeft": 5,
            "marginTop": 30,
            "width": 40,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Editer le profil
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Proposer un pari
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Paris en cours
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Historique des paris
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Regarder une pub pour gagner des pièces
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Contacter Everybet
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FC8D8D",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Se déconnecter
      </Text>
    </View>
  </View>
</View>
`;
